e2e/launchertester/basic_setup_test.go

```diff
@@ -18,20 +20,73 @@ func TestBasicSetup(t *testing.T) {
 	// TODO: try to inject user/password to stdin to avoid --root arg.
 	out, err := launcherCommand(ctx, "install", "--root").CombinedOutput() // Installing as root to avoid Stdin
 	require.NoErrorf(t, err, "Unexpected error installing: %s\n%v", out, err)
-	// TODO: check with Carlos if this is necessary
-	require.NotEmpty(t, out, "Failed to install the distro: No output produced.")
 
 	testCases := map[string]func(t *testing.T){
-		// TODO: Re-enable those tests once the latest wsl-setup with distro patching land.
-		// "SystemdEnabled":          testSystemdEnabled,
-		// "SystemdUnits":            testSystemdUnits,
-		// "CorrectUpgradePolicy":    testCorrectUpgradePolicy,
-		// "UpgradePolicyIdempotent": testUpgradePolicyIdempotent,
-		"InteropIsEnabled": testInteropIsEnabled,
-		"HelpFlag":         testHelpFlag,
+		"SystemdEnabled":          testSystemdEnabled,
+		"SystemdUnits":            testSystemdUnits,
+		"CorrectUpgradePolicy":    testCorrectUpgradePolicy,
+		"UpgradePolicyIdempotent": testUpgradePolicyIdempotent,
+		"InteropIsEnabled":        testInteropIsEnabled,
+		"HelpFlag":                testHelpFlag,
 	}
 
 	for name, tc := range testCases {
 		t.Run(name, tc)
 	}
 }
+
+// TestSetupWithCloudInit runs a battery of assertions after installing with the distro launcher and cloud-init.
+func TestSetupWithCloudInit(t *testing.T) {
+	testCases := map[string]struct {
+		install_root bool
+		wantUser     string
+		wantFile     string
+	}{
+		"With default user in conf":    {wantUser: "testuser", wantFile: "/etc/with_default_user.done"},
+		"Without default user in conf": {wantUser: "testuser", wantFile: "/home/testuser/with_default_user.done"},
+		"Without checking user":        {install_root: true, wantUser: "root", wantFile: "/home/testuser/with_default_user.done"},
```

Comment:
It’s not in this test per say, but there is probably one when there is a default user set in the registry, with or without wsl.conf.

Reply:
That was tricky. I'm still afraid of races, but let's see what CI thinks of my approach to test that.